consumer/publish_event.go

```
@@ -90,6 +92,16 @@ func (e ContentMessage) Matches(whitelist *regexp.Regexp) bool {
 	return whitelist.MatchString(e.ContentURI)
 }
 
+// HasDeleteFlag - checks if payload has delete flag
+func (e ContentMessage) HasDeleteFlag() bool {
+	data, ok := e.Payload.(map[string]interface{})
```

This works however I dislike the hidden code duplication that it introduces.

We have the following piece of code within the `MapNotification` function:

```
if event.HasEmptyPayload() || event.HasDeleteFlag() {
	eventType = dispatch.ContentDeleteType
	contentType = resolveTypeFromMessageHeader(event.ContentTypeHeader)
} else {
	eventType = dispatch.ContentUpdateType
	notificationPayloadMap, ok := event.Payload.(map[string]interface{})
	if ok {
		title = getValueFromPayload("title", notificationPayloadMap)
		contentType = getValueFromPayload("type", notificationPayloadMap)
		scoop = getScoopFromPayload(notificationPayloadMap)
	}
}
```
 
Each received message results in a call to `HasDeleteFlag()` which would do the type assertion we see here. However, if we aren't dealing with delete event we will do the same assertion again as seen in the quoted snippet.

The better approach here would be to extract the type assertion prior to checking for the event type and if it fails, we should return an error. You can follow the changes [here](https://github.com/Financial-Times/content-exporter/pull/32/files) for a similar example.